waddrmgr/manager_test.go

```diff
@@ -1117,6 +1117,202 @@ func testChangePassphrase(tc *testContext) bool {
 	return true
 }
 
+// testNewAccount tests the new account creation func of the address manager works
+// as expected.
+func testNewAccount(tc *testContext) bool {
+	// Cannot create new accounts in watching-only mode
+	if tc.watchingOnly {
+		return true
+	}
+	// Unlock the wallet to decrypt cointype keys required
```

Comment:
Before unlocking the wallet, this needs to verify that attempting to create a new account when the wallet is locked returns `waddrmgr.ErrLocked`.